yb-voyager/cmd/live_migration.go

```diff
@@ -88,10 +125,61 @@ func handleEvent(event *tgtdb.Event) error {
 	if err != nil {
 		return fmt.Errorf("error transforming event key fields: %v", err)
 	}
-	batch := []*tgtdb.Event{event}
-	err = tdb.ExecuteBatch(batch)
-	if err != nil {
-		return fmt.Errorf("error executing batch: %v", err)
-	}
+
+	h := hashEvent(event)
+	evChans[h] <- event
 	return nil
 }
+
+// Returns a hash value between 0..NUM_TARGET_EVENT_CHANNELS
+func hashEvent(e *tgtdb.Event) int {
+	hash := fnv.New64a()
+	hash.Write([]byte(e.SchemaName + e.TableName))
+
+	keyColumns := make([]string, 0)
+	for k := range e.Key {
+		keyColumns = append(keyColumns, k)
+	}
+
+	// sort to ensure input to hash is consistent.
+	sort.Strings(keyColumns)
+	for _, k := range keyColumns {
+		hash.Write([]byte(*e.Key[k]))
+	}
+	return int(hash.Sum64() % (uint64(NUM_TARGET_EVENT_CHANNELS)))
+}
+
+func processEvents(chanNo int, evChan chan *tgtdb.Event, done chan bool) {
+	endOfProcessing := false
+	for !endOfProcessing {
+		batch := []*tgtdb.Event{}
+		timeThreshold := time.After(time.Duration(MAX_TIME_PER_BATCH) * time.Millisecond)
```

Comment:
From the documentation:
__The underlying Timer is not recovered by the garbage collector until the timer fires.__

Consider using `timer := NewTimer(<same-arg-as-time-after>)`; then in the select statement read the `timer.C`; and after the `Batching for` loop call `timer.Stop()`.